src/domains/access.ml

```diff
@@ -356,29 +356,29 @@ and distribute_access_exp f w r c = function
     distribute_access_exp f w     r c e
   | _ -> ()
 
-let add side e w conf vo oo p =
+let add side e w conf mhp vo oo p =
   let ty = get_val_type e vo oo in
   (* let loc = !Tracing.current_loc in *)
   (* ignore (printf "add %a %b -- %a\n" d_exp e w d_loc loc); *)
   match vo, oo with
-  | Some v, Some o -> add_struct side e w conf ty (Some (v, remove_idx o)) p
+  | Some v, Some o -> add_struct side e w conf mhp ty (Some (v, remove_idx o)) p
   | _ ->
     if !unsound && isArithmeticType (type_from_type_offset ty) then
-      add_struct side e w conf ty None p
+      add_struct side e w conf mhp ty None p
     else
-      add_propagate side e w conf ty None p
+      add_propagate side e w conf mhp ty None p
 
 
 (* Access table as Lattice. *)
-(* (varinfo ->) offset -> type -> partition option -> 2^(confidence, write, loc, e, locks) *)
+(* (varinfo ->) offset -> type -> partition option -> 2^(confidence, mhp, write, loc, e, locks) *)
 module A =
 struct
   include Printable.Std
-  type t = int * bool * CilType.Location.t * CilType.Exp.t * LSSet.t [@@deriving eq, ord]
+  type t = int * MHP.t * bool * CilType.Location.t * CilType.Exp.t * LSSet.t [@@deriving eq, ord]
 
-  let hash (conf, w, loc, e, lp) = 0 (* TODO: never hashed? *)
+  let hash (conf, mhp, w, loc, e, lp) = 0 (* TODO: never hashed? *)
 
-  let pretty () (conf, w, loc, e, lp) =
+  let pretty () (conf, mhp, w, loc, e, lp) = (* TODO:Print MHP? *)
```

Comment:
I think this `pretty` isn't really used for user output, but I am wondering of the MHP information should somehow be in the `print_accesses` below, because it could be confusing to debug if certain races are excluded by information which isn't shown.

Reply:
I added printing here now.